### ISSUE TITLE
Disappointing behaviour with multiple monitors

### ISSUE THREAD

**Issue by Keziolio** [Unvanquished/Unvanquished#785]
Hi, i have a 3 monitor system, if i start the game, it goes fullscreen on the first monitor on the left, it's uncomfortable to play.

The game should go fullscreen on the center monitor (at least make a selection menu to switch the display), or span on the 3 monitors (it's better if you give the option to choose).

Now it's annoying to make the game windowed and move it around every time.
Thanks

**Comment by DolceTriade** [comment]
3 monitor support as in have the game stretch across all the monitors? Seems like this is working as expected though.

**Comment by Kangz** [comment]
There is the relatively unknown r_displayIndex that should be doing what you are looking for. I'd be interested in knowing what other games use as a GUI for this kind of options.

**Comment by Keziolio** [comment]
@DolceTriade Yes, but for me it does not work, i see 1920*1080 as the maximum resolution, i'm on linux.

@Kangz source games do have this option in the gui for this

**Comment by DolceTriade** [comment]
We don't support stretching across monitors. I don't think Source games support this either.

**Comment by Keziolio** [comment]
> We don't support stretching across monitors.
Will you?

> I don't think Source games support this either.
They don't

**Comment by DolceTriade** [comment]
We have no plans to support stretching across monitors. I can't imagine a good use case. however, we're always open to contributions from the community to implement the niche features.

**Comment by Keziolio** [comment]
>  I can't imagine a good use case.

Like... playing on all the screens?

**Comment by DolceTriade** [comment]
Yeah, but you have huge bars in between.

**Comment by IngarKCT** [comment]
This is not a daemon issue, in linux it's the window manager that decides where windows appear. In windows games always seem to appear on the primary monitor.

**Comment by Keziolio** [comment]
> Yeah, but you have huge bars in between.

http://img-9gag-fun.9cache.com/photo/aQ4Gqpe_700b.jpg

> This is not a daemon issue, in linux it's the window manager that decides where windows appear. In windows games always seem to appear on the primary monitor.

Yeah, i've already worked around this, it actually appears on the primary monitor

**Comment by Tomtom-mygit** [comment]
> http://img-9gag-fun.9cache.com/photo/aQ4Gqpe_700b.jpg

This. Although I do not have multiple monitors, I think it's a good feature to have in the game – especially to enjoy the wide FOV.

**Comment by DolceTriade** [comment]
Probably a bad example because you would not get the wide FOV, but rather a stretched game. I believe apps using sdl1 (tremulous) will stretch across the available screens to see how it will look.

**Comment by Tomtom-mygit** [comment]
:(

**Comment by Keziolio** [comment]
Here's how it looks on xonotic: http://i.imgur.com/wvSS3lz.jpg

If the game is done right the ui doesn't stretch, and you select the fov

**Comment by DolceTriade** [comment]
Is xonotic using SDL1 or SDL2 in that screenshot. iirc, xonotic can use either.

**Comment by Keziolio** [comment]
That was a screenshot from the git version of darkplaces, i'm not sure but I think is sdl2.
The base version doesn't support it.

Unvanquished is sdl2, isn't it?

**Comment by DolceTriade** [comment]
Yes, Unvanquished is sdl2, though xonotic can be either.

**Comment by TheJJ** [comment]
This really needs to be improved. On nvidia with two monitors, the game first uses the big canvas over both monitors, so it's stretched. After I lowered the resolution to `1920x1080`, it got smaller, but the image is now centered at the gap of both monitors.
Please just use the windowed mode of SDL2, and add a "fullscreen" option to the graphics menu. This shouldn't change the video mode, just create a fullscreen window.

**Comment by DolceTriade** [comment]
By default, we use the desktop resolution for the game in fullscreen. I will mess around with multi monitor setups later this week to see how they can be improved.

**Comment by IngarKCT** [comment]
From the point of view of the application, there is no such thing as multi-monitor. It just gets a render context that has a resolution like 5760x1080. Interestingly enough, with unchanged FOV, this actually means you see less, because of the 48:9 aspect.The stretching is unavoidable because the rendering is still a planar projection, increasing FOV adds more deformation.What the game can do, as suggested, is add an option to render the UI only on the middle part of the context (1/3 width) or do this automaticly depending on the aspect and/or resolution of the render context.

Any remarks about fullscreen modes and resolution are totally irrelevant to this discussion
(a desktop fullscreen would be nice though, swapping out of Unvanquished currently SUCKS)

Note that, with two monitors, the gap is always in the middle. Doing funky stuff like forcing a 1920x1080 resolution on a 3840x1080 device context solves nothing.

Disclamer: I use dual monitor at work, and triple monitor at home.

**Comment by slipher** [comment]
> Now it's annoying to make the game windowed and move it around every time.

We now have a `setWindowOrigin` command so you could potentially automate it...

> The main problem is that this is a "hard fullscreen", which i cannot exit or unfullscreen except by exiting the game.
> 
> Please just use the windowed mode of SDL2, and add a "fullscreen" option to the graphics menu. This shouldn't change the video mode, just create a fullscreen window.

> (a desktop fullscreen would be nice though, swapping out of Unvanquished currently SUCKS)

With the SDL3 migration, we now use desktop/windowed fullscreen rather than true fullscreen.
